### PR TITLE
Added World Time

### PR DESCRIPTION
**Pull request by OfficialKris**
## Description
Added world time. Does not change global lighting. Added corresponding time command and all subfeatures.

Future work (under different PR):
- Allow using time enum under time set command

## Testing
- Tested duration of full day
- Tested changing time
- Tested adding time
- Tested all query commands
- Compared with vanilla time command functionality

## Checklist
Things need to be done before this Pull Request can be merged.

- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings `cargo clippy`

### CODE REVIEW

File: pumpkin/src/entity/player.rs [hunk 1]
```diff
@@ -436,6 +436,10 @@ impl Player {
 
         self.living_entity.tick();
 
+        // TODO: maybe find better way then queuing for just sending data
+        let level_time = self.living_entity.entity.world.level_time.lock().await;
+        level_time.send_time(&self.client).await;
```
Comment (Snowiiii) [inline pumpkin/src/entity/player.rs:441]: Do this in world not in the Player. Also when looking at vanilla code it does
```java
 if (this.ticks % 20 == 0) {
    this.sendTimeUpdatePackets(serverWorld);
}
```

Comment (OfficialKris) [reply pumpkin/src/entity/player.rs:441]: all resources online said it sent the update every tick. I thought it was every second. where does this code come from? is there a link to documentation?

**Comment by OfficialKris** [comment]
Anything left for this PR?

**Review by DataM0del — APPROVED** [review]

File: pumpkin/src/command/commands/cmd_time.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        };
+        // logic
+        let world = server
+            .worlds
+            .first()
+            .expect("There should always be atleast one world");
+        let mut level_time = world.level_time.lock().await;
+        level_time.set_time(time_count.into());
+        let mut owned_string: String = "Change time to: ".to_owned();
+        owned_string.push_str(&time_count.to_string());
```
Comment (Snowiiii) [inline pumpkin/src/command/commands/cmd_time.rs:63]: Just use the `format!` macro

Comment (DataM0del) [reply pumpkin/src/command/commands/cmd_time.rs:63]: So:
```suggestion
        let mut owned_string: String = format!("Change time to: {time_count}");
```

**Comment by OfficialKris** [comment]
Should be all set now

**Comment by Snowiiii** [comment]
Could you please follow the Vanilla /time rules. See https://minecraft.wiki/w/Commands/time
There are 
- `time add`
- `time set`
- `time query`
Should be not hard to implement

**Comment by OfficialKris** [comment]
Time command should be all set now

**Comment by Snowiiii** [comment]
Overall looks good.
Thanks @OfficialKris

**Pull request merged**